### PR TITLE
add 'enable_email_goody' config and fix wording

### PR DESCRIPTION
**Pull request by xuwhite**

### CODE REVIEW

File: resources/views/pages/registration.twig [hunk 1]
```diff
@@ -193,10 +193,11 @@
         <div class="mb-5">
             <h2>{{ __('registration.event_data') }}</h2>
             <div class="row">
-                {% if isGoodieEnabled %}
+                {% if isGoodieEnabled and config('enable_email_goody') %}
```
Comment (MyIgel) [inline resources/views/pages/registration.twig:196]: Imho that check should be done in the controller

File: resources/views/pages/settings/profile.twig [hunk 2]
```diff
@@ -133,9 +133,10 @@
                 {{ f.checkbox('email_human', __('settings.profile.email_by_human_allowed'), {
                     'checked': userdata.settings.email_human,
                 }) }}
-                {% if goodie_enabled %}
+                {% if goodie_enabled and config('enable_email_goody') %}
```
Comment (MyIgel) [inline resources/views/pages/settings/profile.twig:136]: (same here and it has side effects on the saving logic too)

File: config/config.default.php [hunk 3]
```diff
@@ -88,6 +88,10 @@
     # Your privacy@ contact address
     'privacy_email' => env('PRIVACY_EMAIL', null),
 
+    # Enable checkbox to consent for storage of nick, email address and worked hours (and T-shirt size)
```
Comment (MyIgel) [inline config/config.default.php:91]: I would prefer a shorter text here as its currently a copy of the whole translation

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: config/config.default.php (hunk 3 above)
Comment (xuwhite) [reply config/config.default.php:91]: do you have suggestions

Comment (MyIgel) [reply config/config.default.php:91]:
```suggestion
    // Show opt in to long term saving of personal data on user profile and registration pages
```

File: src/Controllers/SettingsController.php [hunk 4]
```diff
@@ -46,7 +46,8 @@ public function profile(): Response
             [
                 'settings_menu' => $this->settingsMenu(),
                 'userdata' => $user,
-                'goodie_enabled' => $this->config->get('goodie_type') !== GoodieType::None->value,
+                'goodie_enabled' => ($this->config->get('goodie_type') !== GoodieType::None->value)
```
Comment (MyIgel) [inline src/Controllers/SettingsController.php:49]:
```suggestion
                'goodie_enabled' => $this->config->get('goodie_type') !== GoodieType::None->value
```

File: src/Controllers/RegistrationController.php [hunk 5]
```diff
@@ -99,7 +99,7 @@ private function renderSignUpPage(): Response
                 'isPasswordEnabled' => $this->userFactory->determineIsPasswordEnabled(),
                 'isDECTEnabled' => $this->config->get('enable_dect'),
                 'isShowMobileEnabled' => $this->config->get('enable_mobile_show'),
-                'isGoodieEnabled' => $goodieType !== GoodieType::None,
+                'isGoodieEnabled' => ($goodieType !== GoodieType::None) && config('enable_email_goody'),
```
Comment (MyIgel) [inline src/Controllers/RegistrationController.php:102]:
```suggestion
                'isGoodieEnabled' => $goodieType !== GoodieType::None && config('enable_email_goody'),
```
does not spark joy ;)

**Pull request merged**